### PR TITLE
add MKLDNNAddtoLayer

### PR DESCRIPTION
**Pull request by tensor-tang**
add mkldnn_addto layer without bias

### CODE REVIEW

**Comment by tensor-tang** [comment]
After merge latest code, still failed of lod_tensor_test 


> [02:19:18] :	 [Step 1/1] The following tests FAILED:
> [02:19:18] :	 [Step 1/1] 	 98 - lod_tensor_test (Failed)
> [02:19:18]W:	 [Step 1/1] Errors while running CTest
> [02:19:18]W:	 [Step 1/1] Process exited with code 8


>  lod_tensor_test .............................***Failed    1.15 sec
[02:14:02] :	 [Step 1/1] Running main() from gtest_main.cc
[02:14:02] :	 [Step 1/1] [==========] Running 6 tests from 2 test cases.
[02:14:02] :	 [Step 1/1] [----------] Global test environment set-up.
[02:14:02] :	 [Step 1/1] [----------] 5 tests from LoDTensorTester
[02:14:02] :	 [Step 1/1] [ RUN      ] LoDTensorTester.NumLevels
[02:14:02] :	 [Step 1/1] unknown file: Failure
[02:14:02] :	 [Step 1/1] C++ exception with description "std::bad_alloc" thrown in SetUp().

**Comment by tensor-tang** [comment]
After rerun, it pass.

File: paddle/gserver/tests/test_MKLDNN.cpp [hunk 1]
```diff
@@ -271,20 +271,53 @@ TEST(MKLDNNLayer, BatchNormLayer) {
...
+    img_conf->set_img_size(pm.iw);
+  }
+}
+
+void testAddtoLayer(const testImageDesc& pm, const size_t nInputs) {
+  CHECK_GE(nInputs, 1);
+  TestConfig dnnConfig;
+  getAddtoConfig(dnnConfig, pm, nInputs);
+  dnnConfig.layerConfig.set_type("mkldnn_addto");
+  // TODO(TJ): test with bias
```
Comment (luotao1) [inline paddle/gserver/tests/test_MKLDNN.cpp]: 下一个PR是完成有bias的addtoLayer？

File: paddle/gserver/tests/test_MKLDNN.cpp [hunk 2]
```diff
@@ -271,20 +271,53 @@ TEST(MKLDNNLayer, BatchNormLayer) {
...
+  for (auto withBias : {false}) {
+    if (withBias) {
+      dnnConfig.biasSize = pm.ic * pm.ih * pm.iw;
+    } else {
+      dnnConfig.biasSize = 0;
```
Comment (luotao1) [inline paddle/gserver/tests/test_MKLDNN.cpp]: dnnConfig.biasSize = withBias？ pm.ic * pm.ih * pm.iw ： 0；
下次改Bias的时候一起改了吧。

File: paddle/gserver/tests/MKLDNNTester.cpp [hunk 3]
```diff
@@ -132,7 +132,7 @@ void MKLDNNTester::checkForward() {
   VLOG(MKLDNN_TESTS) << "Check Forward";
   printTopDatas();
   double delta =
-      compareMatrix(dnnLayer_->getOutputValue(), refLayer_->getOutputValue());
+      compareMatrix(refLayer_->getOutputValue(), dnnLayer_->getOutputValue());
```
Comment (luotao1) [inline paddle/gserver/tests/MKLDNNTester.cpp]: 这里为什么要互换下位置呢

Comment (tensor-tang) [reply paddle/gserver/tests/MKLDNNTester.cpp]: 哦，理论上是可以不要调换的，但是还是想把顺序统一下，并且我默认第一个参数是参考。

File: paddle/gserver/tests/test_MKLDNN.cpp (hunk 1 above)
Comment (tensor-tang) [reply paddle/gserver/tests/test_MKLDNN.cpp]: 嗯，不是下一个就是再下一个了。可能先把Resnet调通下，因为Resnet里面可能还用不到bias的addto。

出于功能，也会实现带有bias的，我已经加到我们list的TODO里面了。

File: paddle/gserver/tests/test_MKLDNN.cpp (hunk 2 above)
Comment (tensor-tang) [reply paddle/gserver/tests/test_MKLDNN.cpp]: 哦好的，thx

**Pull request merged**